rust/arrow/src/buffer.rs

```diff
@@ -774,16 +756,16 @@ impl MutableBuffer {
     /// `new_len` will be zeroed out.
     ///
     /// If `new_len` is less than `len`, the buffer will be truncated.
-    pub fn resize(&mut self, new_len: usize) {
+    pub fn resize(&mut self, new_len: usize, value: u8) {
```

Comment:
we could have another method with `value` = 0 since this seems to be the dominate use case (it might enable future optimizations for the special case as well).

Reply:
I based this signature on `Vec::resize`. I.e. I am assuming here that we use `MutableBuffer` as a special `Vec<u8>` that uses our custom allocator, and offer users an API that they are used to (`Vec`).

The main use-case for `value: 255u8` is when we want to allocate and grow a set bitmap to later unset bits. We did not offer that before, which is the reason `MutableArrayData` does not do that (even though is most likely more efficient for many use-cases).

Note that even though rust `std::RawVec` offers a `with_capacity_zeroed`, `std::Vec` does not expose that as a public API: `resize` and `resize_with` are the public APIs to resize a vec, and none of them uses `std::alloc::alloc_zeroed`, i.e. they also just call `std::ptr::write_bytes`.

I agree with you that `resize(len, 0)` can potentially be optimized when the mutable buffer has zero capacity by calling `std::alloc::alloc_zeroed` instead of `std::ptr::write_bytes`.